cap-primitives/src/fs/mod.rs

```diff
@@ -51,8 +53,10 @@ pub(crate) use unlink_via_parent::*;
 cfg_if::cfg_if! {
     if #[cfg(any(unix, target_os = "fuchsia"))] {
         pub(crate) use super::yanix::fs::*;
+        pub use super::yanix::fs::dir_options::dir_options;
```

Comment:
Ah, I forgot that `dir_options` was in `cap-primitives` while `_open_dir` is in `cap-std`. It feels like `dir_options` and the fact that we use `open` to open directories is an implementation detail of `cap-primitives`. Rather than having `cap-primitives` publish `dir_options` here, could we add an `open_dir` function to `cap-primitives`?

Reply:
Hmm, sounds fair. Lemme see what I can come up with!

Reply:
OK, I've now created a new module in `cap-primitives::fs::open_dir` which is essentially a wrapper around `open` with `dir_options` pre-specified as `OpenOptions` and reused that inside `cap-std` and `cap-async-std`.